### PR TITLE
LE Audio: Add TMAP shell

### PR DESCRIPTION
**Pull request by Thalley**
Add initial TMAP shell module

### CODE REVIEW

File: subsys/bluetooth/audio/shell/tmap.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	if (argc > 1) {
+		shell_error(sh, "%s unknown parameter: %s", argv[0], argv[1]);
+	} else {
+		shell_error(sh, "%s missing subcomand", argv[0]);
+	}
+
+	return -ENOEXEC;
+}
+
+#define HELP_NONE "[none]"
```
Comment (larsgk) [inline subsys/bluetooth/audio/shell/tmap.c:113]: Please consider adding a short help description even though there are no args.

File: subsys/bluetooth/audio/shell/tmap.c [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+static int cmd_tmap_init(const struct shell *sh, size_t argc, char **argv)
+{
+	const enum bt_tmap_role role = (TMAP_CG_SUPPORTED ? BT_TMAP_ROLE_CG : 0U) |
+				       (TMAP_CT_SUPPORTED ? BT_TMAP_ROLE_CT : 0U) |
+				       (TMAP_UMS_SUPPORTED ? BT_TMAP_ROLE_UMS : 0U) |
+				       (TMAP_UMR_SUPPORTED ? BT_TMAP_ROLE_UMR : 0U) |
+				       (TMAP_BMS_SUPPORTED ? BT_TMAP_ROLE_BMS : 0U) |
+				       (TMAP_BMR_SUPPORTED ? BT_TMAP_ROLE_BMR : 0U);
+	int err;
+
```
Comment (larsgk) [inline subsys/bluetooth/audio/shell/tmap.c:53]: Consider adding some text output on the role combination.  Can be a later PR

**Review by larsgk — DISMISSED** [review]
LGTM.  It could be nice with a bit more info in the console for the user but can be another PR

**Comment by Thalley** [comment]
Will update PR based on comments from @larsgk - Definitely need to add some documentation and more

**Review by larsgk — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**